### PR TITLE
Run publish maven snapshots on all branches matching pattern

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Updates the branches that `publish-maven-snapshots` github workflow is run with. It now uses a regex to match any released of unreleased branches. I noticed that the .NET client was using the same patterns: https://github.com/opensearch-project/opensearch-net/blob/main/.github/workflows/test-jobs.yml#L16-L17

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/7705

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwperks** [comment]
Does this need a CHANGELOG entry?

File: .github/workflows/publish-maven-snapshots.yml [hunk 1]
```diff
@@ -5,8 +5,8 @@ on:
   push:
       branches:
         - main
-        - '1.3'
-        - 2.x
+        - '[0-9]+.[0-9]+'
```
Comment (reta) [inline .github/workflows/publish-maven-snapshots.yml:8]: I think it won't hurt: we usually don't backport anything to older branches, so we shouldn't be flooding the repository with unneeded snapshots

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
> Does this need a CHANGELOG entry?

I think no, no code changes

File: .github/workflows/publish-maven-snapshots.yml [hunk 2]
```diff
@@ -5,8 +5,8 @@ on:
...
+        - '[0-9]+.x'
```
Comment (gaiksaya) [inline .github/workflows/publish-maven-snapshots.yml:9]: We also want to be publishing via `2.minor` branch example, 2.7, 2.8, etc.

Comment (cwperks) [reply .github/workflows/publish-maven-snapshots.yml:9]: Line 8 will cover that

**Comment by cwperks** [comment]
@gaiksaya @reta Is there anything outstanding for this PR? It should be good to go.

**Comment by reta** [comment]
> @gaiksaya @reta Is there anything outstanding for this PR? It should be good to go.

Not from me, @gaiksaya LGTM?

**Comment by dblock** [comment]
@gaiksaya Are you saying hard-code that list? That seems alright by me. That list would be true for more than a year.

File: .github/workflows/publish-maven-snapshots.yml [hunk 3]
```diff
@@ -5,8 +5,9 @@ on:
   push:
       branches:
         - main
-        - '1.3'
+        - 1.3
         - 2.x
+        - 2.8
```
Comment (reta) [inline .github/workflows/publish-maven-snapshots.yml]: @gaiksaya 1.3 is fine, but for every new 2.x release would need make a changes once again (2.9, ...), I bilieve @cwperks origin approach was the right one.

Comment (cwperks) [reply .github/workflows/publish-maven-snapshots.yml]: I also prefer the original approach - what is the risk if a snapshot artifact in an old published branch is out-of-sync with the published artifacts? Its pretty rare that changes are merged into branches that have been published except for the most recent published version.

Comment (reta) [reply .github/workflows/publish-maven-snapshots.yml]: I agree, yet another benefit

Comment (gaiksaya) [reply .github/workflows/publish-maven-snapshots.yml]: Works for me. I received some push back in my previous PR. If no concerns by the maintainers we can move forward as is.

**Comment by gaiksaya** [comment]
> @gaiksaya Are you saying hard-code that list? That seems alright by me. That list would be true for more than a year.

Yeah it is hard coded but branches like `2.7`, `2.8` needs to be updated regularly so was recommending the version increment automation to accommodate this change as well. Looks like the current approach with regex works as well?

**Review by gaiksaya — APPROVED** [review]

**Comment by gaiksaya** [comment]
Adding above labels to backport to 2.x, 2.8 and 1.3 branches for now. 
Thanks!

**Pull request merged**